modules/nextflow/src/main/groovy/nextflow/cli/CmdRun.groovy

```diff
@@ -306,6 +306,15 @@ class CmdRun extends CmdBase implements HubOptions {
         // check DSL syntax in the config
         launchInfo(config, scriptFile)
 
+        // Warn about setting NXF_ environment variables within env config scope
+        final nxfEnvs = config.env.findAll { it ->
+            it.toString().startsWith('NXF_') &&
+            ! it.toString().startsWith('NXF_DEBUG')
+        }
+        if( nxfEnvs.size() > 0 ) {
+            log.warn 'Environment variables in the `env` config scope are not applied to Nextflow -- See https://www.nextflow.io/docs/latest/config.html#scope-env'
+        }
+
```

Comment:
ok, let's move this into separate method that takes the config as an argument, and make a unit test for it